source/slang/preprocessor.cpp

```diff
@@ -835,11 +835,15 @@ static Token AdvanceToken(Preprocessor* preprocessor)
 
         // Now re-lex the input
 
+        SourceManager* sourceManager = preprocessor->getCompileRequest()->getSourceManager();
+
         // We create a dummy file to represent the token-paste operation
-        SourceFile* sourceFile = preprocessor->getCompileRequest()->getSourceManager()->allocateSourceFile("token paste", sb.ProduceString());
+        SourceFile* sourceFile = sourceManager->newSourceFile("token paste", sb.ProduceString());
```

Comment:
Style nit: I'd favor `create` over `new` in most cases because it ensures the function name is a verb instead of a noun phrase. I'm actually not clear on why I had `allocate` for the existing function, so maybe there is context here I'm not remembering/understanding.